toolset/utils/docker_helper.py

```diff
@@ -210,7 +210,11 @@ def watch_container(docker_container, docker_file):
     @staticmethod
     def __stop_container(container):
         try:
+            client = container.client
             container.kill()
+            while container.id in map(
+                    lambda x: x.id, client.containers.list()):
+                pass
```

Comment:
Wait, does this work? I thought that we had a problem using `list` in that it would still have stopped containers in it.

Reply:
I think the problem we had before was the opposite. We were checking list too fast for a started container.

Reply:
In other words, we were requesting that the docker daemon start a container, and then checking the list immediately to see that the container had started, and sometimes it wasn't in the list yet. 

This works in that, if the container is not in the list now, it's definitely stopped.